gnomad/gnomadv4_extract_test.py

```diff
@@ -0,0 +1,50 @@
+# LT 23/08/2022
+
+## Extract a small subset of 10 transcripts from gnomADv4
+## annotate with frequencies
+## export the rows as a hail.Table
+
+
+# target transcripts for panel of initial 10 genes selected by the structural constraint project
+# TODO should be in an external file
+
+TRANSCRIPTS = [
+    "ENST00000366684", # ACTA1
+    "ENST00000371953",  # PTEN
+    "ENST00000298910", # LRRK2
+    "ENST00000320868", # HBA1
+    "ENST00000359125", # KCNQ2
+    "ENST00000393087", # SERPINA1
+    "ENST00000281708",  # FBXW7
+    "ENST00000640769", # RTTN
+    "ENST00000270139", # IFNAR1
+    "ENST00000262030" # ATP5F1B
+]
+
+# connect to hail, using appropriate requester_pays setup
+import hail as hl
+hl.init(default_reference='GRCh38',
+        spark_conf = {
+            'spark.hadoop.fs.gs.requester.pays.mode': 'CUSTOM',
+            'spark.hadoop.fs.gs.requester.pays.buckets': 'gnomad',
+            'spark.hadoop.fs.gs.requester.pays.project.id': 'constraint-232598',
+        })
+
+# gnomAD v4 exomes VDS
+big_vds = hl.vds.read_vds('gs://gnomad/v4.0/raw/exomes/gnomad_v4.0.vds')
+
+# find genomics coordinates of target transcripts and filter vds
+target_intervals = hl.experimental.get_gene_intervals(transcript_ids=TRANSCRIPTS, reference_genome='GRCh38')
+small_vds = hl.vds.filter_intervals(big_vds, target_intervals, split_reference_blocks=False, keep=True)
```

Comment:
Why not `split_reference_blocks=True`?

Reply:
would be more computationally intensive with no extra benefit (saving space by segmenting reference blocks is not beneficial here, given the densification next line)

Reply:
Won't densification still give you a larger interval though if you don't split here?

Reply:
No, since we keep only variants.

Reply:
Where does that filtering happen? Why doesn't this also retain ref-blocks? Did you mean to only access `vds.variant_data`?

Reply:
The filtering happens during `densify` which removes all the ref-blocks

Reply:
Are you sure that's the case? E.g.:

```
>>> mt = hl.read_matrix_table('gs://cpg-tob-wgs-test/mt/v7.mt')
>>> mt = mt.head(1000)
>>> mt = hl.experimental.vcf_combiner.densify(mt)
>>> mt.row_key.show()
+---------------+------------+
| locus         | alleles    |
+---------------+------------+
| locus<GRCh38> | array<str> |
+---------------+------------+
| chr1:1        | ["N"]      |
| chr1:10001    | ["T"]      |
| chr1:10003    | ["A"]      |
| chr1:10004    | ["C"]      |
| chr1:10005    | ["C"]      |
| chr1:10006    | ["C"]      |
| chr1:10007    | ["T"]      |
| chr1:10008    | ["A"]      |
| chr1:10009    | ["A"]      |
````

As far as I understand, `densify` fills in hom-refs, but doesn't remove rows.

Reply:
Yes sure, `densify`  removes the REF blocks, the rows were removed earlier by `filter_intervals`. I do not care if there is residual REF/REF rows since I am only interested in getting AN, AC, AF